tools/engine_tool/lib/src/commands/build_command.dart

```diff
@@ -0,0 +1,87 @@
+import 'package:engine_build_configs/engine_build_configs.dart';
+import 'package:path/path.dart' as path;
+import 'package:process_runner/process_runner.dart';
+
+import '../build_utils.dart';
+
+import 'command.dart';
+
+const String _configFlag = 'config';
+
+// TODO(johnmccutchan): Should BuildConfig be BuilderConfig and GlobalBuild be BuildConfig?
+// TODO(johnmccutchan): List all available build targets and allow the user
+// to specify which one(s) we should build on the cli.
+// TODO(johnmccutchan): Can we show a progress indicator like 'running gn...'?
+
+/// The root 'build' command.
+final class BuildCommand extends CommandBase {
+  /// Constructs the 'query' command.
+  BuildCommand({
+    required super.environment,
+    required Map<String, BuildConfig> configs,
+  }) {
+    builds = runnableBuilds(environment, configs);
+    // Add options here that are common to all queries.
+    argParser.addOption(
+      _configFlag,
+      abbr: 'c',
+      defaultsTo: 'host_debug',
+      help: 'Specify the build config to use',
+      allowed: <String>[
+        for (final GlobalBuild config in runnableBuilds(environment, configs))
+          config.name,
+      ],
+      allowedHelp: <String, String>{
+        for (final GlobalBuild config in runnableBuilds(environment, configs))
+          config.name: config.gn.join(' '),
+      },
+    );
+  }
+
+  // List of compatible builds.
+  late final List<GlobalBuild> builds;
+
+  @override
+  String get name => 'build';
+
+  @override
+  String get description => 'Builds the engine';
+
+  @override
+  Future<int> run() async {
+    String configName = argResults![_configFlag] as String;
+    GlobalBuild? build = builds
+        .where((GlobalBuild build) => build.name == configName)
+        .firstOrNull;
+    if (build == null) {
+      throw UnsupportedError('Could not find config ${configName}');
+    }
+    final GlobalBuildRunner buildRunner = GlobalBuildRunner(
+        platform: environment.platform,
+        processRunner: environment.processRunner,
+        abi: environment.abi,
+        engineSrcDir: environment.engine.srcDir,
+        build: build);
+    void handler(RunnerEvent event) {
+      switch (event) {
+        case RunnerStart():
+          environment.logger.info('$event: ${event.command.join(' ')}');
+        case RunnerProgress(done: true):
+          environment.logger.info(event);
+        case RunnerProgress(done: false):
+          {
+            final String percent = '${event.percent.toStringAsFixed(1)}%';
+            final String fraction = '(${event.completed}/${event.total})';
+            final String prefix = '[${event.name}] $percent $fraction ';
+            final String what = event.what;
+            environment.logger.info('$prefix$what');
+          }
```

Comment:
As written, this will scroll a bunch. Alternately, to do something more like ninja currently does, this might work:

```suggestion
        case RunnerProgress(done: true):
          environment.logger.clearLine();
          environment.logger.info(event);
        case RunnerProgress(done: false):
          {
            final String percent = '${event.percent.toStringAsFixed(1)}%';
            final String fraction = '(${event.completed}/${event.total})';
            final String prefix = '[${event.name}] $percent $fraction ';
            final String what = event.what;
            environment.logger.clearLine();
            environment.logger.info('$prefix$what\r', newline: false);
          }
```

Reply:
Done.